index.js

```diff
@@ -45,22 +45,28 @@ function updateBuildStatus(updateConfig, callback) {
 }
 
 /**
- * Shutdown workers and then exit the process
+ * Shutdown both worker and scheduler and then exit the process
  * @method shutDownWorker
  * @param  {Object}       worker    worker to be ended
+ * @param  {Object}       schduler  schduler to be ended
  */
-function shutDownWorker(worker) {
-    worker.end((err) => {
-        if (err) {
-            winston.error(`failed to end the worker: ${err}`);
-            process.exit(128);
+function shutDownAll(worker, scheduler) {
+    worker.end((error) => {
+        if (error) {
+            winston.error(`failed to end the worker: ${error}`);
         }
 
-        process.exit(0);
+        scheduler.end((err) => {
+            if (err) {
+                winston.error(`failed to end the scheduler: ${err}`);
+                process.exit(128);
+            }
+            process.exit(0);
```

Comment:
Doesn't this mean if worker fails to end, but as long as scheduler ends successfully, it still `exit(0)`?

Reply:
yes, but we are not doing anything about the exit code. So this actually doesn't matter